### DISCUSSION TITLE
Raw SQL query inside transaction

### DISCUSSION THREAD

**Discussion by IlyaSemenov** [romeerez/orchid-orm#131] [Q&A] [answered]
Is there a way to run raw SQL queries (such as `create table`) inside `testTransaction` (or `db.$transaction(() => ...)`)?

Consider this:

```ts
await testTransaction.start(db)

await db.$adapter.query(`
  create table "user" (
    id serial primary key,
    name text not null
  );
`)

await db.user.create({ name: "Alice" })
await db.user.take()

await testTransaction.close(db)
```

### Expected

`user` table is rolled back.

### Actual result

`user` table is not rolled back, only `db.user` operations are executed inside the transaction:

```
(10.2ms) BEGIN
(4.3ms) INSERT INTO "user"("name") VALUES ($1) RETURNING * ['Alice']
(0.6ms) SELECT * FROM "user" LIMIT 1
(33.9ms) ROLLBACK
```

**Comment by romeerez** [comment]
Looks like no, because `await db.$adapter.query` won't automatically use the transaction.

I'll need to add a `db.$query` to both make this little bit more compact and to use the transaction automatically.

**Reply by IlyaSemenov** [reply]
That would be great!

**Comment by IlyaSemenov** [comment, marked as answer]
This has been addressed in https://github.com/romeerez/orchid-orm/issues/133
